haystack_experimental/core/super_component/super_component.py

```diff
@@ -5,248 +5,246 @@
-            return comp_name, socket_name
-        except ValueError:
-            raise InvalidMappingError(f"Invalid path format: {path}. Expected 'component_name.socket_name'")
+        Splits a component path into a component name and a socket name.
 
-    def _resolve_input_types(self) -> Dict[str, Dict[str, Any]]:
+        :param path: A string in the format "component_name.socket_name".
+        :returns:
+            A tuple containing (component_name, socket_name).
+        :raises InvalidMappingError:
+            If the path format is incorrect.
         """
-        Resolve input types from pipeline and set them on the component.
+        comp_name, socket_name = parse_connect_string(path)
+        if socket_name is None:
+            raise InvalidMappingError(f"Invalid path format: '{path}'. Expected 'component_name.socket_name'.")
+        return comp_name, socket_name
 
-        Examines the pipeline's declared inputs and merges them with the input mapping
-        to check for type conflicts and set appropriate input types on the wrapper component.
-
-        :returns: The resolved input type data according to auto-resolution or input mapping.
-        :raises InvalidMappingError: If type conflicts are found or mapped components/sockets
-            don't exist
-        """
-        pipeline_inputs = self.pipeline.inputs(include_components_with_connected_inputs=False)
-
-        if input_mapping := self.input_mapping or self.pipeline.metadata.get("input_mapping"):
-            return self._handle_explicit_input_mapping(pipeline_inputs, input_mapping)
-        else:
-            return self._handle_auto_input_mapping(pipeline_inputs)
-
-    def _handle_explicit_input_mapping(  # noqa: PLR0912
+    def _validate_input_mapping(
         self, pipeline_inputs: Dict[str, Dict[str, Any]], input_mapping: Dict[str, List[str]]
-    ) -> Dict[str, Dict[str, Any]]:
+    ) -> None:
         """
-        Handle case where explicit input mapping is provided.
+        Validates the input mapping to ensure that specified components and sockets exist in the pipeline.
 
-        :param pipeline_inputs: Dictionary of pipeline input specifications
-        :param input_mapping: Mapping from wrapper inputs to pipeline socket paths
-        :returns: The resolved input type data resolved according to the input mapping.
-        :raises InvalidMappingError: If mapping is invalid or type conflicts exist
+        :param pipeline_inputs: A dictionary containing pipeline input specifications.
+        :param input_mapping: A dictionary mapping wrapper input names to pipeline socket paths.
+        :raises InvalidMappingError:
+            If the input mapping is invalid or contains nonexistent components or sockets.
         """
-        aggregated_inputs: Dict[str, Dict[str, Any]] = {}
         for wrapper_input_name, pipeline_input_paths in input_mapping.items():
             if not isinstance(pipeline_input_paths, list):
```

Comment:
Would it help if we check the `input_mapping` format as well?
```suggestion
    if not isinstance(input_mapping, dict):
            raise InvalidMappingError("Input mapping must be a dictionary")
        
        for wrapper_input_name, pipeline_input_paths in input_mapping.items():
            if not isinstance(pipeline_input_paths, list):
```

Reply:
Yes, good point!